### ISSUE TITLE
ResidueDB not thread safe.

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#905]
e.g. parsing of AASequences containing modifications with multiple threads causes segmentation faults.
Reason is that modified residues are implicitly added to the ResidueDB singleton if they don't exist.
#5  0x00007ffff6b95586 in OpenMS::Map<OpenMS::String, OpenMS::Map<OpenMS::String, OpenMS::Residue*> >::operator[](this=0x7678e8, key=...)

```
at /abi-data/sachsenb/OpenMS_IDE/src/openms/include/OpenMS/DATASTRUCTURES/Map.h:127
```
#6  0x00007ffff6b92179 in OpenMS::ResidueDB::addResidue_ (this=0x7670b0, r=0x7fffdc004400) at /abi-data/sachsenb/OpenMS_IDE/src/openms/source/CHEMISTRY/ResidueDB.cpp:161
#7  0x00007ffff6b94aae in OpenMS::ResidueDB::getModifiedResidue (this=0x7670b0, residue=0x846070, modification=...)

```
at /abi-data/sachsenb/OpenMS_IDE/src/openms/source/CHEMISTRY/ResidueDB.cpp:483
```
#8  0x00007ffff6b5b00a in OpenMS::AASequence::setModification (this=0x7fffdc004ad0, index=16, modification=...)

```
at /abi-data/sachsenb/OpenMS_IDE/src/openms/source/CHEMISTRY/AASequence.cpp:902
```
#9  0x00007ffff70a8be9 in OpenMS::ModifiedPeptideGenerator::applyFixedModifications (fixed_mods_begin=..., fixed_mods_end=..., peptide=...)

```
at /abi-data/sachsenb/OpenMS_IDE/src/openms/source/ANALYSIS/RNPXL/ModifiedPeptideGenerator.cpp:71
```
#10 0x0000000000497534 in SimpleSearchEngine::main_ (.omp_data_i=0x7fffffffaf40) at /abi-data/sachsenb/OpenMS_IDE/src/utils/SimpleSearchEngine.cpp:582
#11 0x00007ffff400ae3a in gomp_thread_start (xdata=<value optimized out>) at ../.././libgomp/team.c:115
#12 0x00000033854079d1 in start_thread () from /lib64/libpthread.so.0
#13 0x00000030bd2e8b6d in clone () from /lib64/libc.so.6

**Comment by hroest** [comment]
Related to #957

**Comment by hendrikweisser** [comment]
Can this be closed based in #3993 or not yet?
